config/webpack/partial/pwa.js

```diff
@@ -0,0 +1,149 @@
+
+function getSWConfig() {
+  var swConfig;
+
+  try {
+    swConfig = require(swConfigPath);
+  } catch(err) {
+    swConfig = {};
+  }
+
+  return swConfig;
+}
+
+/**
+ * Takes a file path and returns a webpack-compatible
+ * filename descriptor matching the current naming schema
+ * @param  {string} filepath  original file path
+ * @return {string}           parsed file path
+ */
+function getHashedPath(filepath) {
+  var parsed = path.parse(filepath);
+  var name = parsed.name;
+  var ext = parsed.ext;
+  return name + '.[hash]' + ext;
+}
+
+/**
+ * Takes an array of strings representing scripts that
+ * a service worker will import and returns an object
+ * representing the webpack entry config. Each script
+ * will get its own entry point so they can be independantly
+ * imported by the service worker.
+ * @param  {array<string>} importScripts
+ * @param  {object}        current entry config
+ * @return {object}        new entry config
+ */
+function createEntryConfigFromScripts(importScripts, entry) {
+  // Handle the case where there might already be multiple
+  // entry points. If it is we create a new object with all
+  // existing entry points to avoid mutating the config. If its not,
+  // we assume its a string and use it as the bundle entry point.
+  var newEntry = typeof entry === "object"
+    ? Object.assign({}, entry)
+    : { bundle: entry };
+  return importScripts.reduce(function(acc, script) {
+    var name = path.parse(script).name;
+    acc[name] = script;
+    return acc;
+  }, newEntry);
+}
+
+module.exports = function () {
+  return function (config) {
+    var swConfig = getSWConfig();
+
+    if (!swConfig.manifest) {
+      return mergeWebpackConfig(config, {});
+    }
+
+    var manifestConfig = assign({
+      background: "#FFFFFF",
+      logo: "./images/electrode.png",
+      title: "Electrode",
+      statsFilename: "../server/iconstats.json"
+    }, swConfig.manifest);
+
+    var cacheConfig = assign({
+      staticFileGlobs: [
+        "dist/js/*.{js,css}"
+      ],
+      stripPrefix: "dist/js/",
+      cacheId: "electrode",
+      filepath: "dist/sw.js",
+      maximumFileSizeToCacheInBytes: 4194304,
+      skipWaiting: false
+    }, swConfig.cache);
+
+    if (cacheConfig.runtimeCaching) {
+      cacheConfig.runtimeCaching = cacheConfig.runtimeCaching.map(function(runtimeCache) {
+        return {
+          handler: runtimeCache.handler,
+          urlPattern: new RegExp(runtimeCache.urlPattern)
+        }
+      });
+    }
+
+    /**
+     * If importScripts exists in the cache config we need to overwrite
+     * the entry config and output config so we get an entry point for each
+     * script with unique names.
+     */
+    var entry = config.entry;
+    var output = config.output;
+    if (cacheConfig.importScripts) {
+      var importScripts = cacheConfig.importScripts;
+      cacheConfig.importScripts = importScripts.map(getHashedPath);
```

Comment:
This depends on https://github.com/goldhand/sw-precache-webpack-plugin/pull/26 getting merged. Right now `sw-precache-webpack-plugin` is just passing in the filename (`sw.[hash].js`) instead of the full path (`js/sw.[hash].js`)